src/main/java/Helper/UniqueIdGenerator.java

```diff
@@ -0,0 +1,64 @@
+package Helper;
+
+import DBAccesser.User.UserAccessor;
+import DBAccesser.Chat.ChatAccessor;
+import DBAccesser.Message.MessageAccessor;
+
+import java.util.Random;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public final class UniqueIdGenerator {
+
+    @Autowired
+    private UserAccessor queryUser;
+
+    @Autowired
+    private ChatAccessor queryChat;
+
+    @Autowired
+    private MessageAccessor queryMessage; 
+    
+    //generates long type unique Id value for the given table and its corres Id attribute
+    public long generateUniqueId(String tableName, boolean zeroAllowed, boolean negativeAllowed) {
+
+        long id;
+        Random random = new Random();
+
+        while (true) {
+            id = random.nextLong();
+
+            if (zeroAllowed == false && id == 0) {
+                continue;
+            }
+
+            if (negativeAllowed == false && id < 0) {
+                continue;
+            }
+
+            switch (tableName) {
+                case "User": 
+                    if (queryUser.checkIfUserIdExists(id)) {
+                        continue;
+                    } else {
+                        return id;
+                    }
```

Comment:
```
if (!queryUser.checkIfUserIdExists(id)) {
  return id;
}
```

Reply:
But, won't we require a `continue` statement in either case, to prevent _fall through_ in the switch case?

Reply:
You can just use `break`

Reply:
Done